src/realm/object-store/results.cpp

```diff
@@ -258,7 +260,16 @@ util::Optional<T> Results::try_get(size_t ndx)
         }
         if (ndx < m_collection->size()) {
             using U = typename util::RemoveOptional<T>::type;
-            auto mixed = m_collection->get_any(ndx);
+            Mixed mixed;
+            if (m_dictionary_keys) {
```

Comment:
This seems like something that should be done for `get` and `get_any`? The .NET SDK uses `get_any` to iterate the elements of a Results and we're getting the values rather than the keys.

Reply:
Fixed `get_any`. `get` uses `try_get` so that should be fine.